tests/suppression/test_suppression.py

```diff
@@ -37,3 +37,12 @@ def test_filter_messages(self):
             pros = Prospector(config)
             pros.execute()
             self.assertEqual(0, pros.summary["message_count"])
+
+    def test_filter_messages_negative(self):
+        workdir = Path(__file__).parent / "testdata/test_filter_messages_negative"
+        with patch("setoptconf.source.commandline.sys.argv", ["prospector"]):
```

Comment:
Checkout the `tests/utils.py` module, `patch_execution` and `patch_cli` might do what you need here and might as well reuse it

Reply:
@christokur I'm not sure you took carlio's comment into account, could you look into it please ?

Reply:
I took it into account yes
``` 
with patch("setoptconf.source.commandline.sys.argv", ...
```
and 
```
with patch("sys.argv",  ...
```
have different semantics which is why I did not use patch_cli

Reply:
Ok refactored things but be aware that pyupgrade also made changes to how typing is used